src/libraries/System.Runtime.InteropServices.JavaScript/tests/System.Runtime.InteropServices.JavaScript.UnitTests/System/Runtime/InteropServices/JavaScript/WebWorkerTest.Http.cs

```diff
@@ -82,6 +82,7 @@ public async Task HttpClient_CancelInDifferentThread(Executor executor1, Executo
                 {
                     CancellationTokenSource cts = new CancellationTokenSource();
                     var promise = response.Content.ReadAsStringAsync(cts.Token);
+                    Console.WriteLine("HttpClient_CancelInDifferentThread: ManagedThreadId: " + Environment.CurrentManagedThreadId + " NativeThreadId: " + WebWorkerTestHelper.NativeThreadId);
```

Comment:
use `ThrowsAnyAsync<OperationCanceledException>` on line 81 may hide the problem.
The question is why we have 2 different code paths.

Please log stack traces in both cases, so that we could compare them.

Reply:
I kept `ThrowsAsync<TaskCanceledException>`, so issue will still happen but added logs to see stack traces for both `TaskCanceledException` and `OperationCanceledException`.